src/azul/service/avro_pfb.py

```diff
@@ -674,54 +685,71 @@ def _update_replica_schema(*,
     except KeyError:
         schema[key] = _new_replica_schema(path=path, value=value)
     else:
-        if old_type == []:
-            schema[key] = _new_replica_schema(path=path, value=value)
-        elif value is None:
-            if old_type == 'null' or isinstance(old_type, list):
+        if isinstance(old_type, list):
+            _update_replica_schema_union(schema=schema, path=path, key=key, value=value)
+        else:
+            if value is None and old_type == 'null':
                 pass
+            elif isinstance(value, list) and isinstance(old_type, dict) and old_type['type'] == 'array':
+                for v in value:
+                    _update_replica_schema_union(schema=old_type,
+                                                 path=path,
+                                                 key='items',
+                                                 value=v)
+            elif isinstance(value, dict) and isinstance(old_type, dict) and old_type['type'] == 'record':
```

Comment:
Maybe shortening it with something like this… 
```suggestion
            def old_type_is_of_type(type):
                return isinstance(old_type, dict) and old_type['type'] == type
            if value is None and old_type == 'null':
                pass
            elif isinstance(value, list) and old_type_is_of_type('array'):
                for v in value:
                    _update_replica_schema_union(schema=old_type,
                                                 path=path,
                                                 key='items',
                                                 value=v)
            elif isinstance(value, dict) and old_type_is_of_type('record'):
```
… but you may have a better solution.

Reply:
I prefer to just wrap the lines with parentheses rather than introduce a new function. This code is already hard to follow (and I wrote it :sweat_smile: ), so I figure it's best to keep things direct and inline, and it ends up taking less space once you add the obligatory blank line after the function body.